pkg/console/dashboard_panels.go

```diff
@@ -188,22 +181,58 @@ func initState() error {
 	}
 
 	if serverURL != "" {
-		current.harvesterURL = serverURL
 		os.Setenv("KUBECONFIG", "/var/lib/rancher/k3s/agent/kubelet.kubeconfig")
-		return nil
 	}
 
-	current.isMaster = true
-	ip, err := net.ChooseHostInterface()
+	return nil
+}
+
+func syncHarvesterURL(ctx context.Context, g *gocui.Gui) {
+	// sync url at the beginning
+	doSyncHarvesterURL(g)
+
+	syncDuration := 5 * time.Second
+	ticker := time.NewTicker(syncDuration)
+	go func() {
+		<-ctx.Done()
+		ticker.Stop()
+	}()
+	for range ticker.C {
+		doSyncHarvesterURL(g)
+	}
+}
+
+func doSyncHarvesterURL(g *gocui.Gui) {
+	harvesterURL := getHarvesterURL()
+	g.Update(func(g *gocui.Gui) error {
+		v, err := g.View("url")
+		if err != nil {
+			return err
+		}
+		v.Clear()
+		fmt.Fprintf(v, "Harvester management URL: \n\n%s", harvesterURL)
+		return nil
+	})
+}
+
+func getHarvesterURL() string {
+	cmd := exec.Command("/bin/sh", "-c", `kubectl -n harvester-system get svc harvester -o jsonpath='{.status.loadBalancer.ingress[0].ip}'`)
+	cmd.Env = os.Environ()
+	output, err := cmd.CombinedOutput()
 	if err != nil {
-		return err
+		return "Unavailable"
 	}
-	current.harvesterURL = fmt.Sprintf("https://%s:8443", ip.String())
-	return nil
+
+	harvesterAddress := string(output)
```

Comment:
I am concerned that if the command has unexpected output we might show something odd in the console. Is it possible?